init-tools.cmd

```diff
@@ -11,20 +11,41 @@ set /P BUILDTOOLS_VERSION=< "%~dp0BuildToolsVersion.txt"
 set BUILD_TOOLS_PATH=%PACKAGES_DIR%\Microsoft.DotNet.BuildTools\%BUILDTOOLS_VERSION%\lib
 set INIT_TOOLS_RESTORE_PROJECT=%~dp0init-tools.msbuild
 set BUILD_TOOLS_SEMAPHORE_DIR=%TOOLRUNTIME_DIR%\%BUILDTOOLS_VERSION%
-set BUILD_TOOLS_SEMAPHORE=%BUILD_TOOLS_SEMAPHORE_DIR%\init-tools.completed_
+set BUILD_TOOLS_SEMAPHORE=%BUILD_TOOLS_SEMAPHORE_DIR%\init-tools.completed
 
 :: if force option is specified then clean the tool runtime and build tools package directory to force it to get recreated
 if [%1]==[force] (
   if exist "%TOOLRUNTIME_DIR%" rmdir /S /Q "%TOOLRUNTIME_DIR%"
   if exist "%PACKAGES_DIR%\Microsoft.DotNet.BuildTools" rmdir /S /Q "%PACKAGES_DIR%\Microsoft.DotNet.BuildTools"
 )
 
-:: If semaphore exists do nothing
-if exist "%BUILD_TOOLS_SEMAPHORE%" (
-  echo Tools are already initialized.
+if not exist "%BUILD_TOOLS_SEMAPHORE%" (
+  goto :restorebuildtools
+)
+
+::
+:: The init tools semaphore stores the most-recent git commit hash of this file.
+:: When changes to this file are synced, init-tools will refresh the existing tools.
+::
+git rev-list -1 HEAD init-tools.cmd > %BUILD_TOOLS_SEMAPHORE_DIR%\current
```

Comment:
This assumes that there is `git` always on the path. I do not think it is always the case, in particular for external contributors.